discord/bot.py

```diff
@@ -381,9 +381,10 @@ async def register_commands(
 
         commands = [copy.copy(cmd) for cmd in commands]
 
-        for cmd in commands:
-            to_rep_with = [guild_id] if guild_id is not None else guild_id
-            cmd.guild_ids = to_rep_with
+        if guild_id is not None:
+            for cmd in commands:
+                to_rep_with = [guild_id] if guild_id is not None else guild_id
```

Comment:
No need for a second check.
```suggestion
                to_rep_with = [guild_id]
```